### PR TITLE
[WIP, CI did not pass] nvidia-nsight-systems: new package

### PR DESCRIPTION
**Pull request by scothalverson**
nvidia's nsight systems is a system profiling tool.

### CODE REVIEW

File: var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        arch = platform.uname()[-1]
+        if arch == "aarch64":
+            folders = ["/documentation", "/host-linux-armv8a", "/target-linux-sbsa-armv8"]
+        elif arch == "x86_64":
+            folders = ["/documentation", "/host-linux-x64", "/target-linux-x64"]
+        base_path = ""
+        if os.path.exists("opt/nvidia/nsight-systems-cli/"):
+            base_path = "opt/nvidia/nsight-systems-cli/"
+        elif os.path.exists("opt/nvidia/nsight-systems/"):
+            base_path = "opt/nvidia/nsight-systems/"
```
Comment (junghans) [inline var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py:66]: Shouldn't there be an error if neither exists?

File: var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        for sd in folders:
+            shutil.copytree(base_path + ver + sd, prefix + sd)
+        os.mkdir(prefix + "/bin")
+        os.symlink(prefix + "/host-linux-armv8/nsys-ui", prefix + "/bin/nsys-ui")
+        os.symlink(prefix + "/target-linux-sbsa-armv8/nsys", prefix + "/bin/nsys")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py:72]: Is that `target-linux-sbsa-armv8` part also true for `arch == "x86_64"`?

**Review by bernhardkaindl — COMMENTED** [review]
@scothalverson Would you be able to fix the review comments by @junghans and squash your commits into one commit?

**Comment by scothalverson** [comment]
Yes, sorry - busy couple of weeks, but I will address.

**Comment by rbberger** [comment]
@scothalverson any update on this?

**Comment by scothalverson** [comment]
my initial motivation for creating this was that nsight systems had some new features for ARM cpus, and the version of nsight systems that made it into NVHPC lagged behind the most recent release and did not have the new features.  That issue has been resolved, so users can get what they need from NVHPC.  As a result the priority is lower, but I will fix the above issues.

**Comment by giordano** [comment]
I was about to open an issue to request a new package for nvidia-nsight-systems, but I just found this PR already!

> That issue has been resolved, so users can get what they need from NVHPC. As a result the priority is lower, but I will fix the above issues.

While it's true that Nsight Systems is part of the whole NVHPC bundle, that package is _huge_.  If one needs only the profiler, it'd be much nicer to be able to install only that component instead of the whole megapackage, which takes much more time and storage.  So I'd be happy to see this package in.

File: var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    and GPUs, from large servers to the smallest system on a chip"""
+
+    homepage = "https://developer.nvidia.com/nsight-systems"
+    url = "https://developer.download.nvidia.com/devtools/repos/"
+    maintainers("scothalverson")
+    license("NVIDIA Software License Agreement")
+
+    depends_on("libarchive programs='bsdtar'")
+
+    arch = platform.uname()[-1]
```
Comment (giordano) [inline var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py:27]:
```python
>>> import platform
>>> platform.uname()
uname_result(system='Linux', node='dream', release='6.11.4-amd64', version='#1 SMP PREEMPT_DYNAMIC Debian 6.11.4-1 (2024-10-20)', machine='x86_64')
>>> platform.uname()[-1]
''
```
This looks wrong?

**Comment by giordano** [comment]
I opened #47355 as an alternative to this PR to address all the comments above.

**Comment by scothalverson** [comment]
Hi @giordano!  Thanks for doing this!  I've held off on making modifications to support this due to some internal discussions, but I realize a solution now is useful.  I'm happy to support this when/if there's additional iteration that needs to happen.

**Pull request closed without merging**